frontend/src/pages/documents/documents.page.vue

```diff
@@ -1,27 +1,159 @@
+import { DocumentDto } from '@shared/dto';
+import BaseButton from '@components/buttons/link-button.vue';
+import BaseCard from '@components/cards/card.vue';
+import CreateDialog from '@components/dialogs/create-dialog.vue';
+import DeleteDialog from '@components/dialogs/delete-dialog.vue';
+import ContributorDialog from '@components/dialogs/contributor-dialog.vue';
 
 export default defineComponent({
   components: {
     Page,
-    Document,
+    TheDocument,
+    BaseButton,
+    BaseCard,
+    CreateDialog,
+    DeleteDialog,
+    ContributorDialog,
+  },
+
+  data() {
+    return {
+      createDialogOpened: false,
+      title: '',
+      emptyTitle: false,
+      deleteDialogOpened: false,
+      documentID: '',
+      contributorDialogOpened: false,
+      userName: '',
+      emptyUserName: false,
+    };
   },
 
   computed: {
-    documents() {
+    documents(): DocumentDto[] {
       return documentsVuexModule.documents;
     },
+    hasDocuments(): number {
+      return documentsVuexModule.documents.length;
+    },
+  },
+
+  methods: {
+    toggleCreateDialog(): void {
+      this.createDialogOpened = !this.createDialogOpened;
+      if (!this.createDialogOpened) this.emptyTitle = false;
+    },
+    toggleDeleteDialog(id?: string): void {
+      if (id) this.documentID = id;
+      this.deleteDialogOpened = !this.deleteDialogOpened;
+    },
+    toggleContributorDialog(id?: string): void {
+      if (id) this.documentID = id;
+      this.contributorDialogOpened = !this.contributorDialogOpened;
+      if (!this.contributorDialogOpened) this.emptyUserName = false;
+    },
+    createdDocument(): void | undefined {
+      if (this.title === '') {
+        this.emptyTitle = true;
+        return;
+      }
+      this.emptyTitle = false;
+      this.toggleCreateDialog();
+      const id = this.createID();
+      const currentUser = authVuexModule.currentUser;
+      if (currentUser) {
+        const document = {
+          id: id,
+          title: this.title,
+          user: currentUser,
+          contributors: [currentUser.name],
+          createdDate: this.dateAndTime(),
+          modifiedDate: this.dateAndTime(),
+        };
+        documentsVuexModule.createDocument(document);
+      }
+      this.title = '';
+    },
+    deleteDocument(): void {
+      this.toggleDeleteDialog();
+      const index = documentsVuexModule.documents.findIndex(value => value.id === this.documentID);
+      documentsVuexModule.deleteDocument(index);
+    },
+    addContributor(): void | undefined {
+      if (this.userName === '') {
+        this.emptyUserName = true;
+        return;
+      }
+      this.emptyUserName = false;
+      this.toggleContributorDialog();
+      const index = documentsVuexModule.documents.findIndex(value => value.id === this.documentID);
+      documentsVuexModule.addContributor([index, this.userName]);
+      this.changeModifiedDate(index);
+      this.userName = '';
+    },
```

Comment:
This component is TOO big. It can be mush simplier if you move all validation rules to separate component (InputValidation) and move all data manipulation to vuex. Vuex was specially created for cases when some components are related but you dont want to expand your parent component too much